app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingFragment.kt

```diff
@@ -0,0 +1,364 @@
+            context
+        } else {
+            throw IllegalStateException("Fragment must be attached to a BottomSheetActivity!")
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val arguments = requireArguments()
+
+        viewModel.init()
+
+        isSwipeToRefreshEnabled = arguments.getBoolean(ARG_ENABLE_SWIPE_TO_REFRESH, true)
+
+        val preferences = PreferenceManager.getDefaultSharedPreferences(requireContext())
+        val statusDisplayOptions = StatusDisplayOptions(
+            animateAvatars = preferences.getBoolean(PrefKeys.ANIMATE_GIF_AVATARS, false),
+            mediaPreviewEnabled = accountManager.activeAccount!!.mediaPreviewEnabled,
+            useAbsoluteTime = preferences.getBoolean(PrefKeys.ABSOLUTE_TIME_VIEW, false),
+            showBotOverlay = preferences.getBoolean(PrefKeys.SHOW_BOT_OVERLAY, true),
+            useBlurhash = preferences.getBoolean(PrefKeys.USE_BLURHASH, true),
+            cardViewMode = if (preferences.getBoolean(
+                    PrefKeys.SHOW_CARDS_IN_TIMELINES,
+                    false
+                )
+            ) CardViewMode.INDENTED else CardViewMode.NONE,
+            confirmReblogs = preferences.getBoolean(PrefKeys.CONFIRM_REBLOGS, true),
+            confirmFavourites = preferences.getBoolean(PrefKeys.CONFIRM_FAVOURITES, false),
+            hideStats = preferences.getBoolean(PrefKeys.WELLBEING_HIDE_STATS_POSTS, false),
+            animateEmojis = preferences.getBoolean(PrefKeys.ANIMATE_CUSTOM_EMOJIS, false)
+        )
+        adapter = TrendingPagingAdapter(
+            statusDisplayOptions,
+            this,
+        )
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_trending, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        setupSwipeRefreshLayout()
+        setupRecyclerView()
+
+        adapter.registerAdapterDataObserver(object : RecyclerView.AdapterDataObserver() {
+            override fun onItemRangeInserted(positionStart: Int, itemCount: Int) {
+                if (positionStart == 0 && adapter.itemCount != itemCount) {
+                    binding.recyclerView.post {
+                        if (getView() != null) {
+                            if (isSwipeToRefreshEnabled) {
+                                binding.recyclerView.scrollBy(
+                                    0,
+                                    Utils.dpToPx(requireContext(), -30)
+                                )
+                            } else binding.recyclerView.scrollToPosition(0)
+                        }
+                    }
+                }
+            }
+        })
+
+        if (actionButtonPresent()) {
```

Comment:
Should the FAB be present here? I'm not sure it makes sense (unless we want people to be able to create a new status from anywhere in the app).

Reply:
I thought it should be. This is a bigger question that me though. It seems to be possible everywhere else? 

My opinion, I don't think we lose anything by keeping it.

Reply:
FAB will be hidden